From d6e295061f239bee48c9e49313f68042121e21c2 Mon Sep 17 00:00:00 2001
From: Pavel Begunkov <asml.silence@gmail.com>
Date: Mon, 15 Apr 2024 13:50:13 +0100
Subject: [PATCH] io_uring/notif: shrink account_pages to u32
Git-commit: d6e295061f239bee48c9e49313f68042121e21c2
Patch-mainline: v6.10-rc1
References: bsc#1230569

->account_pages is the number of pages we account against the user
derived from unsigned len, it definitely fits into unsigned, which saves
some space in struct io_notif_data.

Signed-off-by: Pavel Begunkov <asml.silence@gmail.com>
Link: https://lore.kernel.org/r/19f2687fcb36daa74d86f4a27bfb3d35cffec318.1713185320.git.asml.silence@gmail.com
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Gabriel Krisman Bertazi <krisman@suse.de>
---
 io_uring/notif.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/io_uring/notif.h b/io_uring/notif.h
index 52e124a9957c..2e25a2fc77d1 100644
--- a/io_uring/notif.h
+++ b/io_uring/notif.h
@@ -13,7 +13,8 @@
 struct io_notif_data {
 	struct file		*file;
 	struct ubuf_info	uarg;
-	unsigned long		account_pages;
+
+	unsigned		account_pages;
 	bool			zc_report;
 	bool			zc_used;
 	bool			zc_copied;
-- 
2.47.0

